sdk/core/System.ClientModel/src/Pipeline/HttpClientPipelineTransport.Request.cs

```diff
@@ -0,0 +1,193 @@
+using System.Collections.Generic;
+using System.IO;
+using System.Net;
+using System.Net.Http;
+using System.Net.Http.Headers;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace System.ClientModel.Primitives;
+
+public partial class HttpClientPipelineTransport
+{
+    private class HttpPipelineRequest : PipelineRequest
+    {
+        private const string AuthorizationHeaderName = "Authorization";
+
+        private string _method;
+        private Uri? _uri;
+        private BinaryContent? _content;
+
+        private readonly PipelineRequestHeaders _headers;
+
+        private bool _disposed;
+
+        protected internal HttpPipelineRequest()
+        {
+            _method = HttpMethod.Get.Method;
+            _headers = new PipelineRequestHeaders();
+        }
+
+        protected override string GetMethodCore()
+            => _method;
+
+        protected override void SetMethodCore(string method)
+        {
+            Argument.AssertNotNull(method, nameof(method));
+
+            _method = method;
+        }
+
+        protected override Uri GetUriCore()
+        {
+            if (_uri is null)
+            {
+                throw new InvalidOperationException("Uri has not be set on HttpMessageRequest instance.");
+            }
+
+            return _uri;
+        }
+
+        protected override void SetUriCore(Uri uri)
+        {
+            Argument.AssertNotNull(uri, nameof(uri));
+
+            _uri = uri;
+        }
+
+        protected override BinaryContent? GetContentCore()
+            => _content;
+
+        protected override void SetContentCore(BinaryContent? content)
+            => _content = content;
+
+        protected override PipelineMessageHeaders GetHeadersCore()
+            => _headers;
+
+        // PATCH value needed for compat with pre-net5.0 TFMs
+        private static readonly HttpMethod _patchMethod = new HttpMethod("PATCH");
+
+        private static HttpMethod ToHttpMethod(string method)
+        {
+            return method switch
+            {
+                "GET" => HttpMethod.Get,
+                "POST" => HttpMethod.Post,
+                "PUT" => HttpMethod.Put,
+                "HEAD" => HttpMethod.Head,
+                "DELETE" => HttpMethod.Delete,
+                "PATCH" => _patchMethod,
+                _ => new HttpMethod(method),
+            }; ;
+        }
+
+        internal static HttpRequestMessage BuildHttpRequestMessage(PipelineRequest request, CancellationToken cancellationToken)
+        {
+            HttpMethod method = ToHttpMethod(request.Method);
+            Uri uri = request.Uri;
+            HttpRequestMessage httpRequest = new HttpRequestMessage(method, uri);
+
+            MessageBodyAdapter? httpContent = request.Content == null ? null :
+                new MessageBodyAdapter(request.Content, cancellationToken);
+            httpRequest.Content = httpContent;
+#if NETSTANDARD
+            httpRequest.Headers.ExpectContinue = false;
+#endif
```

Comment:
just curious - what does this do?

Reply:
https://github.com/Azure/azure-sdk-for-net/pull/25328